editor/js/SkinnedMuscle.js

```diff
@@ -24,22 +28,35 @@ THREE.SkinnedMuscle = function(geom, points, material) {
     this.material = material;
     this.material.skinning = true;
 };
+
 THREE.SkinnedMuscle.prototype = Object.create( THREE.SkinnedMesh.prototype );
 THREE.SkinnedMuscle.prototype.constructor = THREE.SkinnedMuscle;
+
 THREE.SkinnedMuscle.prototype.updateMatrixWorld = function( force ) {
 // if has pathpoints attribute then it's a muscle
 // Cycle through pathpoints, update their matrixworld
 // then set the position of the Bones from that
     if (this.skeleton === undefined)
         return;
     var bones = this.skeleton.bones;
+    //console.warn("Num bones in updateMatrixWorld: " + bones.length);
+
     if (this.pathpointObjects.length != this.pathpoints.length){
-        for ( var b=0; b < this.pathpoints.length; b++) {
-            var ppt = this.pathpoints[b];
-            var pptObject = editor.objectByUuid(ppt);
-            if (pptObject !== undefined) {
-                this.pathpointObjects.push(pptObject);
-                bones[b].geometry = pptObject.geometry;
+        var b = 0;
+        for ( var p=0; p < this.pathpoints.length; p++) {
+            var pptObject1 = editor.objectByUuid(this.pathpoints[p]);
+            var pptObject2 = editor.objectByUuid(this.pathpoints[p+1]);
+
+            if (pptObject1 !== undefined) {
+               // add every pathpoint to the list of PathPoint objects
+                this.pathpointObjects.push(pptObject1);
+
+                if (pptObject2 !== undefined) {
+                // define the two bones of a segement of the path together
+                    bones[b].geometry = new THREE.SphereGeometry(8, 64, 64); //pptObject1.geometry;
+                    bones[++b].geometry = new THREE.SphereGeometry(8, 64, 64); //pptObject2.geometry;
+                }
```

Comment:
Why create new SphereGeometry here. We reuse whatever the Geometry associated with the pathPoint by copying the reference. I don't think undoing this will change anything in the fix/solution